### PR TITLE
Adding documentation about frontend conventions.

### PR DESCRIPTION
**Pull request by linuspahl**
## Motivation and Context

Before this change, we were maintaining our frontend conventions in our corporate wiki. We decided to move them to our repository instead. The maintenance via git is preferred by many developers and this way it is also accessible for open source users.

With this PR we are also restructuring and extending the convention list, please have a look at the commit history for more information.

### CODE REVIEW

File: graylog2-web-interface/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+## Conventions
+
+### Consistent Code Style
+- We use ESLint to detect some issues in our code. We mostly follow the [Airbnb Javascript style guide](https://github.com/airbnb/javascript) to write frontend code, with a few exceptions.
+- We maintain those custom rules in a package, which is part of [our graylog2-server repository](https://raw.githubusercontent.com/Graylog2/graylog2-server/master/graylog2-web-interface/packages/eslint-config-graylog/index.js).
+- Ensure you are seeing linter hints in you IDE and consider enabling the setting "fix linter hints on save" ([IntelliJ Docs](https://www.jetbrains.com/help/idea/eslint.html#ws_eslint_configure_run_eslint_on_save)).
+- There is a CI job, which checks if there are linter hints in changes files.
+- If you want to see all linter hints for changed (and committed) files locally, you can run `yarn lint:changes`. You will need to commit your changes first.
+- If you want to see linter hints for a specific file, you can run `lint:path /path/to/file`.
```
Comment (kmerz) [inline graylog2-web-interface/CONTRIBUTING.md:23]:
```suggestion
- If you want to see linter hints for a specific file, you can run `yarn lint:path /path/to/file`.
```

File: graylog2-web-interface/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+### Class vs functional components
+Small components should be functional components. When a component is more complex, you can decide which type of component
+you want to use (class or a functional component with react hooks). When you don’t have a preference, use a functional component.
+
+### Type definitions
+- We use TypeScript for new components, and we also define `PropType` definitions. Static types add better type support for the development, integrating with your IDE, while `PropType` definitions are present at runtime.
+- Names of unused parameters
+  - Sometimes you need to define a parameter which is not being used, for example when you just want to use the second parameter. To satisfy TypeScript the parameter name needs to be prefixed with underscore.
+  - Instead of defining just `_`, use a more meaningful name, like `_theParameterName`. (Related to this discussion){.
+- types.d.ts can have hidden errors, like missing imports. Rename it temporary to `types.d.ts` to detect the most obvious ones.
```
Comment (kmerz) [inline graylog2-web-interface/CONTRIBUTING.md:41]: renaming `types.d.ts` to `types.d.ts`?? I don't get it.

Comment (linuspahl) [reply graylog2-web-interface/CONTRIBUTING.md]: Thanks, it should be "rename `types.d.ts` to `types.ts`".

File: graylog2-web-interface/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+- Please ensure to check the [frontend documentation](https://graylog2.github.io/frontend-documentation) to see which common components we use. Whenever possible try using a common component, since that help us have a more consistent UI and make components used in different parts of the product behave the same way.
+
+### Internal packages
+We currently have a few internal packages, used for the core application and all plug-ins:
+- graylog-web-plugin contains common packages for both the core and plug-ins, webpack configuration for building plug-ins, as well as some interfaces to register and consume plugins.
+- eslint-config-graylog contains our custom linter configuration, based on eslint-config-airbnb.
+- stylelint-config-graylog contains our custom stylelint configuration, based on the default stylelint config.
+
+### Browser compatibility
+We currently do not have a pre release process to test different or old browser. Nevertheless you should test layout changes in at least all modern browsers (Chrome, Firefox, Safari). Bigger layout changes should also be tested in older browsers. Have a look at our public browser compatibility list.
```
Comment (zeeklop) [inline graylog2-web-interface/CONTRIBUTING.md:133]: A link to the list would be nice

File: graylog2-web-interface/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+### UI Styling
+
+#### Forms
+- Latest research suggest forms with aligned labels and inputs are the easiest to use and navigate for users, so avoid using horizontal forms unless there is a good reason for it.
+- Add helper text to help users fill the input.
+- Only show validation state when something changed or the form was submitted.
+- Prefer our own Select component over the default one.
+- Avoid adding long forms into a modal.
+- Mark optional fields as optional by adding “(Optional)” to the label text.
+- We are working in using standard components and libraries to make our forms more consistent.
```
Comment (zeeklop) [inline graylog2-web-interface/CONTRIBUTING.md:174]: Change to `We are working in...` to `We are working on...`

File: graylog2-web-interface/CONTRIBUTING.md (hunk 3 above)
Comment (linuspahl) [reply graylog2-web-interface/CONTRIBUTING.md:133]: Good point.

**Comment by linuspahl** [comment]
Thank you all, I merged this PR for now, if there are further suggestions please create a new PR.

**Pull request merged**